README.md

```diff
@@ -144,33 +96,59 @@ Any plugins you have installed or code you invoke in Sketch's _Plugins > Run Scr
    ```
 1. Start Sketch and your changes will be picked up.
 
-> ⚠️ If you re-build SketchAPI by running `npm run build` again while Sketch is open you won't see your changes automatically reflected. You'll need to restart Sketch for this to happen.
+> ⚠️ Sketch must be restarted for API changes to take effect after running `npm run build`.
 
-> ⚠️ Once you've finished working on SketchAPI don't forget to stop Sketch using your customised version, to do this run:<br/>`npm run api-location:delete`.
+> ⚠️ You must remove the custom location configuration to make your installation of Sketch use the version of the API included with the app.
+>
+> ```
+> npm run api-location:delete
+>
+> ```
 
 ### Testing
 
-The `*.test.js` files in this repository are integration tests that run in Sketch's plugin environment, using the skpm test runner.
+The SketchAPI builds on top of macOS and internal Sketch APIs via CocoaScript. To ensure the API works for a specific Sketch version or build, this repository includes `*.test.js` files containing integration tests.
+
+These integration tests are compiled into a single test plugin using Webpack and can run by the `run_tests.py` Python script, or from the Sketch application menu.
+
+**Build test plugin**
+
+To build the plugin separately, e.g. to install and run it manually, run the following command.
+
+```sh
+npm install # if you haven't installed dependencies already
+npm run test:build --identifier=IDENTIFIER
+```
 
-To run these tests using your current version of the Sketch as the host environment invoke:
+To build the plugin including only one spec file, e.g. run tests from one spec only, run this command:
 
-```bash
-npm run test
+```sh
+npm run test:build --identifier=IDENTIFIER
 ```
 
-Alternatively if you want to run the tests with a specific app binary run:
+Use different `IDENTIFIER` values if you are testing different versions of Sketch or want to run integration tests concurrently on the same machine.
+
+> **Note:** The Sketch JavaScript API does not need to be rebuilt. The integration tests use the API bundled within Sketch or the custom API location specified in the user defaults.
+
+**Run test plugin**
 
-```bash
-SKETCH_PATH=/path/to/sketch.app npm run test
+To run the integration test plugin from the command-line, you must provide the path to the plugin and a file path to be used by the plugin to write the test results and can be watched by the Python script for changes.
+
+```sh
+python run_tests.py -p /path/to/SketchIntegrationTests-$UUID.sketchplugin -o FILE_PATH [-s SKETCH_PATH] [-t TIMEOUT]
```

Comment:
Yeah, which might be related to the slightly unorthodox macOS Python setup. Python 2 has been discontinued since 1 January 2020 and the Python script makes it explicit that it's Python 3.